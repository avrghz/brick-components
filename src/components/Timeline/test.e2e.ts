import { newE2EPage, E2EPage, E2EElement } from '@stencil/core/testing'

describe('<bk-timeline/>', () => {
    let page: E2EPage
    let component: E2EElement

    beforeEach(async () => {
        page = await newE2EPage()
        await page.setContent(`<bk-timeline>
            <bk-timeline-item time="01-01-2020 19:20" variant="success">
                <h4>Update Github template</h4>
                <p>Tom committed 01-01-2020 19:20</p>
            </bk-timeline-item>
            <bk-timeline-item time="01-01-2020 19:35" variant="success">
                <h4>Update Github template</h4>
                <p>Tom committed 01-01-2020 19:35</p>
            </bk-timeline-item>
            <bk-timeline-item time="01-01-2020 19:50" variant="danger">
                <h4>Update Github template</h4>
                <p>Tom committed 01-01-2020 19:50</p>
            </bk-timeline-item>
        </bk-timeline>`)
        component = await page.find('bk-timeline')
    })

    it('should render', async () => {
        expect(component).toMatchSnapshot()
    })
})
