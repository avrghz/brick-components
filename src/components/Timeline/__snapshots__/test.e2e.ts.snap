// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bk-timeline/> should render 1`] = `
<bk-timeline class="hydrated">
  <bk-timeline-item class="hydrated sc-bk-timeline-item-h sc-bk-timeline-item-s" time="01-01-2020 19:20" variant="success">
    <!---->
    <div class="bk-timeline-item sc-bk-timeline-item">
      <div class="bk-timeline-item__tail sc-bk-timeline-item"></div>
      <div class="bk-timeline-item__node bk-timeline-item__node--normal bk-timeline-item__node--success sc-bk-timeline-item"></div>
      <div class="bk-timeline-item__wrapper sc-bk-timeline-item">
        <div class="bk-timeline-item__timestamp is-top sc-bk-timeline-item">
          01-01-2020 19:20
        </div>
        <div class="bk-timeline-item__content sc-bk-timeline-item">
          <bk-card class="bk-card hydrated is-always-shadow sc-bk-card-h sc-bk-card-s sc-bk-timeline-item">
            <!---->
            <div class="sc-bk-timeline-item sc-bk-timeline-item-s" slot="body">
              <h4>
                Update Github template
              </h4>
              <p>
                Tom committed 01-01-2020 19:20
              </p>
            </div>
          </bk-card>
        </div>
      </div>
    </div>
  </bk-timeline-item>
  <bk-timeline-item class="hydrated sc-bk-timeline-item-h sc-bk-timeline-item-s" time="01-01-2020 19:35" variant="success">
    <!---->
    <div class="bk-timeline-item sc-bk-timeline-item">
      <div class="bk-timeline-item__tail sc-bk-timeline-item"></div>
      <div class="bk-timeline-item__node bk-timeline-item__node--normal bk-timeline-item__node--success sc-bk-timeline-item"></div>
      <div class="bk-timeline-item__wrapper sc-bk-timeline-item">
        <div class="bk-timeline-item__timestamp is-top sc-bk-timeline-item">
          01-01-2020 19:35
        </div>
        <div class="bk-timeline-item__content sc-bk-timeline-item">
          <bk-card class="bk-card hydrated is-always-shadow sc-bk-card-h sc-bk-card-s sc-bk-timeline-item">
            <!---->
            <div class="sc-bk-timeline-item sc-bk-timeline-item-s" slot="body">
              <h4>
                Update Github template
              </h4>
              <p>
                Tom committed 01-01-2020 19:35
              </p>
            </div>
          </bk-card>
        </div>
      </div>
    </div>
  </bk-timeline-item>
  <bk-timeline-item class="hydrated sc-bk-timeline-item-h sc-bk-timeline-item-s" time="01-01-2020 19:50" variant="danger">
    <!---->
    <div class="bk-timeline-item sc-bk-timeline-item">
      <div class="bk-timeline-item__tail sc-bk-timeline-item"></div>
      <div class="bk-timeline-item__node bk-timeline-item__node--danger bk-timeline-item__node--normal sc-bk-timeline-item"></div>
      <div class="bk-timeline-item__wrapper sc-bk-timeline-item">
        <div class="bk-timeline-item__timestamp is-top sc-bk-timeline-item">
          01-01-2020 19:50
        </div>
        <div class="bk-timeline-item__content sc-bk-timeline-item">
          <bk-card class="bk-card hydrated is-always-shadow sc-bk-card-h sc-bk-card-s sc-bk-timeline-item">
            <!---->
            <div class="sc-bk-timeline-item sc-bk-timeline-item-s" slot="body">
              <h4>
                Update Github template
              </h4>
              <p>
                Tom committed 01-01-2020 19:50
              </p>
            </div>
          </bk-card>
        </div>
      </div>
    </div>
  </bk-timeline-item>
</bk-timeline>
`;
